src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionXmlFormatReader.cs

```diff
@@ -169,50 +172,14 @@ private void ReflectionReadMember(object obj, int memberIndex, XmlReaderDelegato
             }
         }
 
-        private object ReflectionGetMemberValue(object obj, MemberInfo memberInfo)
+        private object ReflectionGetMemberValue(object obj, DataMember dataMember)
         {
-            object memberValue = null;
-            if (memberInfo is PropertyInfo)
-            {
-                PropertyInfo propInfo = (PropertyInfo)memberInfo;
-                memberValue = propInfo.GetValue(obj);
-            }
-            else if (memberInfo is FieldInfo)
-            {
-                FieldInfo fieldInfo = (FieldInfo)memberInfo;
-                memberValue = fieldInfo.GetValue(obj);
-            }
-            else
-            {
-                throw new InvalidOperationException($"The type, {memberInfo.GetType()}, of memberInfo is not supported.");
-            }
-
-            return memberValue;
+            return dataMember.Getter(obj);
```

Comment:
I would rather not do that as it allows a non-matching MemberInfo to be passed to the wrong Getter method. Creating a closure avoids the possibility of this type of bug.